backend.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type Backend interface {
+	BackendReader
+	BackendWriter
+}
+
+// Implement to store event
+type BackendWriter interface {
+	insert(DomainEntry) error
+	ping(HostEntry) error
+}
+
+// Implement to retrieve event from backend and convert to QueryResponse
+type BackendReader interface {
+	get(string) (QueryResponse, error)
+	update() ([]HostEntry, error)
+}
+
+// distributedBackend
+type distributedBackend struct {
+	hosts       []string
+	persistance Backend
+}
+
+func NewDistributedBackend(
+	hosts []string, persistBackend Backend) *distributedBackend {
+	b := new(distributedBackend)
+	b.hosts = hosts
+	b.persistance = persistBackend
+	return b
+}
+
+func (b distributedBackend) ping(entry HostEntry) error {
+	return b.persistance.ping(entry)
+}
+
+func (b distributedBackend) get(domain string) (QueryResponse, error) {
+	var total int = 0
+	bounced := false
+	resp, err := b.persistance.get(domain)
+	if err != nil {
+		log.Fatalf("Error selecting from persistance: %s", err)
```

Comment:
Just noticed, are you sure we wanna crash the program here? :)

Reply:
Another good catch. The Cassandra connection needs to have reconnect logic added. Also it would be good, given the independent nature of the nodes, to make them less reliant on Cassandra even being up at all. It would make them much more robust. 

They can queue up the tasks to be sent off to the backend when it becomes available again. Though they should definitely log that their backend connection has died.

Having monitoring, grafana or what not, on the nodes for things like this, and memory usage, should exist in a production environment.